### PR TITLE
feat: retrieve invitations by superAdmin

### PR DESCRIPTION
**Pull request by Micah-Shallom**
## Description
Developed an API endpoint to retrieve all invitations sent by an admin for users to join an organization. This endpoint provides the status of each invitation, indicating whether it has been accepted, rejected, or is still pending.

## Related Issue (Link to Github issue)
Issue #109 : https://github.com/hngprojects/hng_boilerplate_golang_web/issues/109#issue-2424092132

## How Has This Been Tested?
The functionality was tested using end to end tests to ensure the correct retrieval of invitations. Tests included:
- Successful retrieval of invitations with valid admin credentials.
- Handling unauthorized access when a non-admin tries to access the endpoint.
- Managing missing authentication tokens.

### Testing Environment:
- Golang 1.18
- Gin Gonic framework
- PostgreSQL for the database
- Testing tools included `httptest` and custom assertion utilities.

## Screenshots (if appropriate - Postman, etc):
Not applicable.
a1

a2

a3

a4

a5

a6

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## How to TEST
- Sign Up as an ADMIN - 127.0.0.1:8019/api/v1/auth/admin/signup
```
{
    "first_name":"test",
    "last_name":"micah",
    "email":"micahtest@gmail.com",
    "username":"mtest",
    "password":"password",
    "phone_number":"09071111150"
}
```
- Use the generated token as Bearer Token
- Create an Organisation
```
{
    "name":"testOrg",
    "description":"an organisation",
    "email":"micahtest@gmail.com",
    "state":"kaduna",
    "industry":"tech",
    "type":"tech",
    "address":"Nigeria",
    "country":"Nigeria"
}
```
- Get the Organisation id from the token
- Create an invitation 
```
{
  "organisation_id": "0190e1f6-3766-7285-858b-d6c55d84930e",
  "email": "micahtest1000@gmail.com"
}

```
- Access the endpoint  /api/v1/organisation/invitations/

### CODE REVIEW

**Review by Cyberguru1 — DISMISSED** [review]
lgtm

**Comment by Micah-Shallom** [comment]
awaiting review @NwokoyeChigozie

File: pkg/controller/invite/postInvite.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+		invitations = append(
+			invitations,
+			map[string]interface{}{
+				"email":        email,
+				"organization": org.Name,
+				"expires_at":   invitation.ExpiresAt.Format(time.RFC3339),
+				// "token":        token,
+			},
+		)
+
```
Comment (NwokoyeChigozie) [inline pkg/controller/invite/postInvite.go]: move these to a service

File: pkg/controller/invite/postInvite.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+		return
+	}
+
+	userClaims := claims.(jwt.MapClaims)
+	userId := userClaims["user_id"].(string)
+
+	// Check if user is a member of the organization
+	isMember, err := organisation.CheckUserIsMemberOfOrg(userId, org.ID, base.Db.Postgresql)
+	if err != nil {
+		rd := utility.BuildErrorResponse(http.StatusNotFound, "error", "User not a member of the organization", err, nil)
```
Comment (NwokoyeChigozie) [inline pkg/controller/invite/postInvite.go]: only request validation should be on your controller, please look at what others did

**Review by NwokoyeChigozie — CHANGES_REQUESTED** [review]

**Comment by Micah-Shallom** [comment]
@NwokoyeChigozie have made changes

**Review by urizennnn — APPROVED** [review]

**Review by Cyberguru1 — APPROVED** [review]
all good

File: pkg/controller/invite/postInvite.go (hunk 2 above)
Comment (urizennnn) [reply pkg/controller/invite/postInvite.go]: .

File: pkg/controller/invite/postInvite.go (hunk 1 above)
Comment (urizennnn) [reply pkg/controller/invite/postInvite.go]: resolved

**Review by urizennnn — DISMISSED** [review]

**Review by Cyberguru1 — DISMISSED** [review]

**Pull request closed without merging**